### PR TITLE
feat(cli) Improve environment variables parsing

### PR DESCRIPTION
**Pull request by Hywan**
# Description

This PR fixes https://github.com/wasmerio/wasmer/issues/2028. In https://github.com/wasmerio/wasmer/pull/1762 we have already improved environment variables syntax, but the same work must be done in the CLI crate.

The following environment variables are considered incorrect:

* `A`, no equal sign,
* `A=`, value is missing,
* `=A`, key is missing.

The following environment variables are considered correct:

* `A=B` with `A` for the name and `B` for the value,
* `A=B=C` with `A` for the name and `B=C` for the value.

The environment variable is trimmed before being parsed with [`str::trim`](https://doc.rust-lang.org/std/primitive.str.html#method.trim).

This PR also adds tests for the `parse_envvar` function, which was missing.

# Review

- [x] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: lib/cli/src/utils.rs [hunk 1]
```diff
@@ -40,14 +40,53 @@ pub fn parse_mapdir(entry: &str) -> Result<(String, PathBuf)> {
...
+            "Environment variable is not well formed, the `name` is missing in `<name>=<value>`; got `{}`",
+            &entry
+        ),
+
+        Some(position) if position == entry.len() - 1 => bail!(
+            "Environment variable is not well formed, the `value` is missing in `<name>=<value>`; got `{}`",
             &entry
+        ),
+
+        Some(position) => Ok((entry[..position].into(), entry[position + 1..].into())),
```
Comment (MarkMcCaskey) [inline lib/cli/src/utils.rs:63]: The actual splitting can just be done with just https://doc.rust-lang.org/std/primitive.str.html#method.split_at ; but we won't have as many error cases to report.

Just a heads up, I'm fine keeping it as-is

**Review by MarkMcCaskey — APPROVED** [review]
Looks good , thanks for the tests!

File: lib/cli/src/utils.rs (hunk 1 above)
Comment (Hywan) [reply lib/cli/src/utils.rs:63]: We would need to deal with empty string for the left- or right-end side of the `=`. I think the code would not be simplified that much. Thanks for the heads-up!

**Pull request merged**